r/R/record-batch.R

```diff
@@ -274,6 +274,12 @@ as.data.frame.RecordBatch <- function(x, row.names = NULL, optional = FALSE, ...
 }
 
 .serialize_arrow_r_metadata <- function(x) {
+  # drop problems attributes (most likely from readr)
+  if ("attributes" %in% names(x) &&
+      "problems" %in% names(x[["attributes"]]) ) {
+    x[["attributes"]][["problems"]] <- NULL
+  }
+
```

Comment:
I believe you can safely just
```suggestion
  x[["attributes"]][["problems"]] <- NULL
```

```
> x <- list()
> x
list()
> x$attributes$problems <- NULL
> x
list()
> x[["attributes"]][["problems"]] <- NULL
> x
list()
```

Reply:
_it turns out_ just `NULL`ing doesn’t quite work with the tests for garbage metadata (`"garbage"[["attributes"]][["problems"]] <- NULL` errors. I can either test that the attributes are a list before this or alter our garbage tests to send in a list.

Reply:
Should we validate/assert `is.list()` in this function?

Reply:
Yeah I can do that. Though [one of the existing tests](https://github.com/apache/arrow/blob/master/r/tests/testthat/test-metadata.R#L76-L81) fails here — the one right above it works which might be sufficient, is there something that lines 76-81 is testing that we aren't covering with lines 70-75?

Reply:
I would have to look into the implementation to be sure, but I think the intent of the first test should be that we don't save garbage `r` metadata and the second would be that, if someone happened to have bad metadata stored in the `r` key (we can't prevent some other data generating process from doing that), that we don't crash on loading it. It's not clear that that's actually what those tests are doing, but that's what I think they should do.

Reply:
Ok, cool — I'll rewrite the second test to avoid using `.serialize_arrow_r_metadata()` so that we don't error there, but still keep the testing-serialized-data aspect of that test.